src/endpoint/sts/ops/sts_post_assume_role.js

```diff
@@ -35,12 +35,13 @@ async function assume_role(req) {
                 Credentials: {
                     AccessKeyId: access_keys.access_key.unwrap(),
                     SecretAccessKey: access_keys.secret_key.unwrap(),
-                    Expiration: expiry,
+                    Expiration: Date.now() + expiry,
                     SessionToken: generate_session_token({
                         access_key: access_keys.access_key.unwrap(),
                         secret_key: access_keys.secret_key.unwrap(),
                         assumed_role_access_key: assumed_role.access_key
-                    }, expiry)
+                    // generate_session_token expects an expiry in seconds
```

Comment:
You can move this comment to a JDoc explanation in `generate_session_token` and rename the `expiry` to `expiry_ms` (in milliseconds) hoping it will be easier to understand.